canonical-web-and-design/maas-ui
# Rework nav and routes + add scaffolding for network forms

## Done
- Reworked Nav so that main section paths are optional (i.e. Network has no path and is therefore not clickable), and subsection paths are the whole path. Subsections are also highlighted in the side nav.
- Updated styling of `/network/proxy` to closer match `/storage`
- Added routes and basic components for `/network/dns`, `/network/ntp`, `/network/syslog` and `/network/network-discovery`
- Rearranged the Storage and Network folders so that all subsections sit under the main nav folders, e.g `containers/Network/ProxyForm`, `containers/Network/DnsForm` instead of `/containers/ProxyForm`, `/containers/DnsForm`

## QA
- Go to /network and check that you get redirected to /network/proxy
- Check that you can change values and save the form, and the values persist after refresh
- Check that saving shows a spinner + success icon
- Check that the sidenav is correctly highlighted for the different network subsections
- Check that the storage form still works properly too

## Screenshot
![Screen Shot 2019-08-13 at 14 57 35-fullpage](https://user-images.githubusercontent.com/25733845/62943413-b513ba80-bdda-11e9-96c2-e61e756210f1.png)
